mscore/qml/palettes/PaletteTree.qml

```diff
@@ -306,22 +306,32 @@ ListView {
         id: typeAheadTimer
         interval: 1000
         onTriggered: typeAheadStr = ''
-        }
+    }
     onTypeAheadStrChanged: {
         if (typeAheadStr.length)
             typeAheadTimer.restart();
     }
 
+    function numberOfExpandedPalettes() {
+        const startIndex = paletteModel.index(0, 0);
+        const collapsedIndexList = paletteModel.match(startIndex, PaletteTreeModel.PaletteExpandedRole, true);
+        return collapsedIndexList.length;
+    }
+
+    function numberOfCollapsedPalettes() {
+        const startIndex = paletteModel.index(0, 0);
+        const collapsedIndexList = paletteModel.match(startIndex, PaletteTreeModel.PaletteExpandedRole, false);
+        return collapsedIndexList.length;
+    }
+
     function expandCollapseAll(expand) {
         console.assert([true, false, null].indexOf(expand) !== -1, "Invalid value for expand: " + expand);
         // expand = true  - expand all
         //          false - collapse all
         //          null  - decide based on current state
         if (expand === null) {
             // if any are collapsed then expand all, otherwise collapse all
-            const startIndex = paletteModel.index(0, 0);
-            const collapsedIndexList = paletteModel.match(startIndex, PaletteTreeModel.PaletteExpandedRole, false);
-            expand = !!collapsedIndexList.length;
+            expand = !!numberOfExpandedPalettes()
```

Comment:
(OK, I see that `!!integerValue` is apparently MuseScore-idiomatic, I take back previous comments)

You did leave out a `;` though.